beeware/briefcase
# Fixes #359 - Adds a Flatpak backend for Linux packaging

Adds a [Flatpak](https://flatpak.org) backend as an alternative for Linux packaging.

To test:
* Install `flatpak` and `flatpak-builder` as Linux system packages. Installation instructions can be found [on the  flatpack website](https://flatpak.org/setup). On most systems, this should be as straightforward as `apt-get install flatpak flatpack-builder` (or whatever the OS-appropriate installer tool is).
* Add the following to the `linux` section of your `pyproject.toml`:

       template = "https://github.com/beeware/briefcase-linux-flatpak-template"
       template_branch = "dev"

* Run `briefcase X linux flatpak` (where X is create, build, run, or package)

This PR doesn't deal with:
* Signing the bundled single-file packages
* Publishing packages to flathub or other repositories.

These would both be possible, but require building an integration with GPG, which is a whole integration unto itself.

Thanks to @loganasherjones for their [extremely helpful blog post](https://www.loganasherjones.com/2018/05/using-flatpak-with-python/), and @nickzoic for an initial draft PR #754.

## PR Checklist:

- [x] All new features have been documented
